mlrun/track/trackers/mlflow_tracker.py

```diff
@@ -0,0 +1,552 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import os
+import pathlib
+import tempfile
+import zipfile
+from typing import List, Union
+
+import mlflow
+import mlflow.entities
+import mlflow.environment_variables
+import mlflow.types
+
+import mlrun
+from mlrun import MLClientCtx, mlconf
+from mlrun.artifacts import Artifact, ModelArtifact
+from mlrun.features import Feature
+from mlrun.launcher.client import ClientBaseLauncher
+from mlrun.model import RunObject
+from mlrun.projects import MlrunProject
+from mlrun.utils import logger, now_date
+
+from ..tracker import Tracker
+
+
+class MLFlowTracker(Tracker):
+    """
+    A class for detecting and logging MLFlow runs into MLRun. Notice, only the last active MLFlow run is logged.
+    """
+
+    def __init__(self):
+        super().__init__()
+        self._pre_runs = None
+        self._experiment_name = None
+        self._run_id = None
+
+    @staticmethod
+    def is_enabled() -> bool:
+        """
+        Checks if tracker is enabled.
+
+        :return: True if the tracking configuration is enabled, False otherwise.
+        """
+        return getattr(mlconf.external_platform_tracking, mlflow.__name__).enabled
+
+    def pre_run(self, context: MLClientCtx):
+        """
+        Initializes the MLFlow tracker, setting the experiment name as configured by the user.
+
+        :param context: Current mlrun context
+        """
+        # Check for a user set experiment name via the environment variable:
+        self._experiment_name = (
+            mlflow.environment_variables.MLFLOW_EXPERIMENT_NAME.get()
+        )
+
+        # Check if the user configured for matching the experiment name with the context name:
+        if getattr(
+            mlconf.external_platform_tracking, mlflow.__name__
+        ).match_experiment_to_runtime:
+            if self._experiment_name is not None:
+                context.logger.warn(
+                    f"`mlconf.external_platform_tracking.mlflow.match_experiment_to_runtime` is set to True but the "
+                    f"MLFlow experiment name environment variable ('MLFLOW_EXPERIMENT_NAME') is set for using the "
+                    f"name: '{self._experiment_name}'. This name will be overriden "
+                    f"with MLRun's runtime name as set in the MLRun configuration: '{context.name}'."
+                )
+            self._experiment_name = context.name
+            mlflow.environment_variables.MLFLOW_EXPERIMENT_NAME.set(
+                self._experiment_name
+            )
+
+        # We have 3 options to track our run, either we set the run id, or we have the experiment, and then we check
+        # for added runs in it, or we do none, and then we look for added runs across all experiments.
```

Comment:
This type of comment will be easier to read
```suggestion
        # We have 3 options to track our run:
        # 1. Set the run id.
        # 2. Use the experiment, and check for added runs in it.
        # 3. Look for added runs across all experiments.
```